pydoctor/epydoc/markup/epytext.py

```diff
@@ -1230,35 +1240,35 @@ def descr(self):
 ##                    SUPPORT FOR EPYDOC
 #################################################################
 
-def parse_docstring(docstring, errors):
+def parse_docstring(docstring: str, errors: List[ParseError]) -> ParsedDocstring:
     """
     Parse the given docstring, which is formatted using epytext; and
-    return a C{ParsedDocstring} representation of its contents.
+    return a L{ParsedDocstring} representation of its contents.
+
     @param docstring: The docstring to parse
-    @type docstring: C{string}
     @param errors: A list where any errors generated during parsing
         will be stored.
-    @type errors: C{list} of L{ParseError}
-    @rtype: L{ParsedDocstring}
     """
     tree = parse(docstring, errors)
     if tree is None:
         return ParsedEpytextDocstring(None, ())
 
+    tree_children = cast(List[Element], tree.children)
```

Comment:
are all these casts needed just because `parse` can return None ?

Reply:
No, `tree.children` has type `List[Union[str, Element]]`, but the root element will never be given children of type `str`.

Maybe at some point we should create separate node subclasses for different kinds of elements, but for now I'm happy just annotating epydoc's original design without changing it much.